### ISSUE TITLE
Do large negative user biases mean anything?

### ISSUE THREAD

**Issue by trygvebw** [lyst/lightfm#521]
I have created a LightFM model that recommends users movies and TV series based on their viewing history. The model works reasonably well, with a test AUC score around 0.75 and a test precision@K=15 score around 0.05. Subjectively speaking, the generated recommendations are fine although a bit more generic/less diverse than I'd ideally want.

I was looking at the user biases, and noticed that they're all large and negative (with a mean around -175) both for users with few and for users with many interactions. The item biases (and user/item embeddings) look "normal", lying in a range between roughly -0.5 and 0.5.

Does this mean anything, e.g. that the model hasn't converged properly?

![image](https://user-images.githubusercontent.com/1441500/75983839-8f224480-5ee9-11ea-91a1-0a680612f74c.png)

**Comment by trygvebw** [comment]
The large user biases vastly dominate all users' prediction scores. Here's an example of the top prediction scores for one user:
![image](https://user-images.githubusercontent.com/1441500/75984111-1a033f00-5eea-11ea-94af-9d3c503554f4.png)

**Comment by maciejkula** [comment]
I agree that it's slightly weird, but as you point out it shouldn't really matter. Perhaps adding a bit of L2 regularization will help keep things more contained?

**Comment by maciejkula** [comment]
Regarding diversity: this is absolutely a common problem. You may have to do some post-processing on the final results to optimize it (this is one possible way: https://dl.acm.org/doi/10.1145/3269206.3272018).
